src/PolarSSL++/BlockingSslClientSocket.cpp

```diff
@@ -189,7 +189,7 @@ bool cBlockingSslClientSocket::Send(const void * a_Data, size_t a_NumBytes)
 	size_t NumBytes = a_NumBytes;
 	for (;;)
 	{
-		int res = m_Ssl.WritePlain(a_Data, a_NumBytes);
+		int res = m_Ssl.WritePlain(Data, a_NumBytes);
```

Comment:
Actually this is wrong. `Data` is the correct variable to use - the `WritePlain` method could consume less than `a_NumBytes` bytes, in which case the loop continues, but only the remaining portion of the data is to be sent (line 202).